ca/server.go

```diff
@@ -808,3 +826,162 @@ func isFinalState(status api.IssuanceStatus) bool {
 
 	return false
 }
+
+func (s *Server) finishRootRotation(expectedRootCA *api.RootCA) error {
+	// when the nodes are all converged on having TLS certificates issued by the correct entity, we can
+	// try to finish the root rotation
+	clusterID := s.securityConfig.ClientTLSCreds.Organization()
+	return s.store.Update(func(tx store.Tx) error {
+		cluster := store.GetCluster(tx, clusterID)
+		if cluster == nil {
+			return fmt.Errorf("unable to get cluster %s", clusterID)
+		}
+
+		// If the RootCA object has changed (because another root rotation was started or because some other node
+		// had finished the root rotation), we cannot finish the root rotation that we were working on.
+		if !equality.RootCAEqualStable(expectedRootCA, &cluster.RootCA) {
+			return errors.New("target root rotation has changed - unable to complete previous root rotation")
+		}
+
+		var signerCert []byte
+		if len(cluster.RootCA.RootRotation.CAKey) > 0 {
```

Comment:
The previous block checks that the root CA is the same as our expected root CA, modulo join tokens.  The expected root CA is what one loop of the `reconcileNodeRootsAndCerts` attempts to converge on, and that loop checks to make sure the root rotation isn't nil.  So if the cluster's root CA is the same as the root CA we are trying to finish a rotation for, then it should have a root rotation that isn't nil.

I should probably add a comment saying that at the top of this function, though.